### PR TITLE
Improved Accordion

### PR DESCRIPTION
**Pull request by navin-moorthy**
- API is based on Reakit's Tab component(Composite & Disclosure)
- Arrow Navigation from the idea of Segment component
- Fixed issues with compose hooks for better accordion usage
- Separate state for simple accordion & allowMultiple accordion.
- Better types with comments & default
- Updated Storybook stories
- Update Tests for the new API
- Handled most of the edge cases for allowToggle & allowMultiple including thier respective wai-aria requirement

### CODE REVIEW

File: src/accordion/AccordionItem.tsx [hunk 1]
```diff
@@ -1,84 +1,150 @@
...
+            //  the current Id in composite hence thats handled directly in state
+            options.select(options.id);
+            return;
+          }
+
+          options.unSelect(options.id);
+          return;
+        }
+
+        options.select?.(options.id);
```
Comment (anuraghazra) [inline src/accordion/AccordionItem.tsx:111]: This bit of code is same as the OnClick handler's, extract it out.

Comment (navin-moorthy) [reply src/accordion/AccordionItem.tsx:111]: I thought you will say this 😆

Comment (anuraghazra) [reply src/accordion/AccordionItem.tsx:111]: Yeah lol 😆 My eyes & brain screams when they see repetitive code which can be refactored :p

File: src/accordion/AccordionPanel.ts [hunk 2]
```diff
@@ -1,55 +1,140 @@
...
+  }
+
+  const panelIndex = getPanelIndex(options.panels, panel);
+  const accordionsWithoutPanel = getAccordionsWithoutPanel(
+    options.items,
+    options.panels,
+  );
+  return accordionsWithoutPanel[panelIndex]?.id || undefined;
+}
+
```
Comment (anuraghazra) [inline src/accordion/AccordionPanel.ts:89]: Move these functions to the end since isPanelVisible is also on the end.

Comment (navin-moorthy) [reply src/accordion/AccordionPanel.ts:89]: Okay 👍

**Review by anuraghazra — APPROVED** [review]
